src/analysis/inner_product_space/adjoint.lean

```diff
@@ -0,0 +1,145 @@
+begin
+  ext v,
+  refine ext_inner_left 𝕜 (λ w, _),
+  rw [adjoint_aux_inner_right, adjoint_aux_inner_left],
+end
+
+@[simp] lemma adjoint_aux_norm (A : E →L[𝕜] F) : ∥adjoint_aux A∥ = ∥A∥ :=
+begin
+  refine le_antisymm _ _,
+  { refine continuous_linear_map.op_norm_le_bound _ (norm_nonneg _) (λ x, _),
+    rw [adjoint_aux_apply, linear_isometry_equiv.norm_map],
+    exact to_sesq_form_apply_norm_le },
+  { nth_rewrite_lhs 0 [←adjoint_aux_adjoint_aux A],
+    refine continuous_linear_map.op_norm_le_bound _ (norm_nonneg _) (λ x, _),
+    rw [adjoint_aux_apply, linear_isometry_equiv.norm_map],
+    exact to_sesq_form_apply_norm_le }
+end
+
+/-- The adjoint of a bounded operator from Hilbert space E to Hilbert space F. -/
+def adjoint : (E →L[𝕜] F) ≃ₗᵢ⋆[𝕜] (F →L[𝕜] E) :=
+linear_isometry_equiv.of_surjective
+{ norm_map' := adjoint_aux_norm,
+  ..adjoint_aux }
+(λ A, ⟨adjoint_aux A, adjoint_aux_adjoint_aux A⟩)
+
+local postfix `†`:1000 := adjoint
+
+/-- The fundamental property of the adjoint. -/
+lemma adjoint_inner_left (A : E →L[𝕜] F) (x : E) (y : F) : ⟪A† y, x⟫ = ⟪y, A x⟫ :=
+adjoint_aux_inner_left A x y
+
+/-- The fundamental property of the adjoint. -/
+lemma adjoint_inner_right (A : E →L[𝕜] F) (x : E) (y : F) : ⟪x, A† y⟫ = ⟪A x, y⟫ :=
+adjoint_aux_inner_right A x y
+
+/-- The adjoint is involutive -/
+@[simp] lemma adjoint_adjoint (A : E →L[𝕜] F) : A†† = A :=
+adjoint_aux_adjoint_aux A
+
+/-- The adjoint of the composition of two operators is the composition of the two adjoints
+in reverse order. -/
+@[simp] lemma adjoint_comp (A : F →L[𝕜] G) (B : E →L[𝕜] F) : (A ∘L B)† = B† ∘L A† :=
+begin
+  ext v,
+  refine ext_inner_left 𝕜 (λ w, _),
+  simp only [adjoint_inner_right, continuous_linear_map.coe_comp', function.comp_app],
+end
+
+/-- The adjoint is unique: a map `A` is the adjoint of `B` iff it satisfies `⟪A x, y⟫ = ⟪x, B y⟫`
+for all `x` and `y`. -/
+lemma eq_adjoint_iff (A : E →L[𝕜] F) (B : F →L[𝕜] E) :
+  (∀ x y, ⟪A x, y⟫ = ⟪x, B y⟫) ↔ A = B† :=
+begin
+  refine ⟨λ h, _, λ h x y, by rw [h, adjoint_inner_left]⟩,
+  ext x,
+  exact ext_inner_right 𝕜 (λ y, by simp only [adjoint_inner_left, h x y])
+end
+
+/-- `E →L[𝕜] E` is a star algebra with the adjoint as the star operation. -/
+instance : has_star (E →L[𝕜] E) := ⟨adjoint⟩
+instance : has_involutive_star (E →L[𝕜] E) := ⟨adjoint_adjoint⟩
+instance : star_monoid (E →L[𝕜] E) := ⟨adjoint_comp⟩
+instance : star_ring (E →L[𝕜] E) := ⟨linear_isometry_equiv.map_add adjoint⟩
+instance : star_module 𝕜 (E →L[𝕜] E) := ⟨linear_isometry_equiv.map_smulₛₗ adjoint⟩
+
+section real
+
+variables {E' : Type*} {F' : Type*} [inner_product_space ℝ E'] [inner_product_space ℝ F']
+variables [complete_space E'] [complete_space F']
+
+lemma is_adjoint_pair (A : E' →L[ℝ] F') :
+  bilin_form.is_adjoint_pair (bilin_form_of_real_inner : bilin_form ℝ E')
+  (bilin_form_of_real_inner : bilin_form ℝ F') A.to_linear_map (A†).to_linear_map :=
```

Comment:
Can you please use
```suggestion
  (bilin_form_of_real_inner : bilin_form ℝ F') A (A†) :=
```
or if Lean needs it to be explicit
```suggestion
  (bilin_form_of_real_inner : bilin_form ℝ F') (A : E' →ₗ[ℝ] F') (A† : E' →ₗ[ℝ] F') :=
```
Coercion ought to behave better than `.to_linear_map` in interaction with `simp`.